kernel_gateway/notebook_http/handlers.py

```diff
@@ -45,11 +47,12 @@ class NotebookAPIHandler(TokenAuthorizationMixin,
     services.cell.parser.APICellParser for detail about how the source cells
     are identified, parsed, and associated with HTTP verbs and paths.
     """
-    def initialize(self, sources, response_sources, kernel_pool, kernel_name):
+    def initialize(self, sources, response_sources, kernel_pool, kernel_name, kernel_language='no_spec'):
```

Comment:
Why are you using a magic value 'no_spec' as the default, instead of null or an empty string? I don't see a check for this magic value in your changes. If this "language" has now a special meaning, it would have to be documented somewhere.

Reply:
I intended to add it as a self-explanatory value, or maybe there's a need of checking on it in further development. I think an empty string is ok, too.